Added "enctype" option, change the request encodings type.

v-zaburdaev opened the pull request:
I add "enctype" option, change the request encodings type.
multipart/form-data by default, and i add urlencoded and json encoding types.
multipart/form-data parsed too slow or our server side application, urlencoded much faster.

v-zaburdaev committed:
Added enctype option, change the request encodings type.

* **enctype** : set POST request encodings type. Default is multipart/form-data. can be "form-data", "urlencoded" or "json"

## How to use "enctype" option?
This option means the request encoding type, by default it uses 'multipart/form-data' encodings,
for multpart/form-data
			enctype: "form-data"
for urlencode
			enctype: "urlencoded"
for json
			enctype: "json"

v-zaburdaev committed:
fix enctype default value and readme

v-zaburdaev committed:
Fixed bug with reading from FormData. It was possible only in chrome.

BenjaminVadant commented:
Hi,

thank you for the pull request, it adds a really interesting feature. 

However, I would like you to make some changes before merging : 
- define and document all the options you add (for instance disableGreenLines, showrawfcddata & afterFetch) 
- use the same type of indentation as the master branch (2 spaces)
- I would prefer that you convert the formdata to a urlencoded string or a json string and not the other way around as the FormData.append method can create arrays and direct assignation can't (around line 45)
- I would prefer not using a jQuery method for url encoding (as it adds a dependency) 

Thanks

Benjamin

v-zaburdaev committed:
Some fixes - removed not used options  disableGreenLines, showrawfcddata.
removed jQuery dependency to create urlencoded string.
Formatting - used 2 spaces indentation.

v-zaburdaev committed:
Added encodeURIComponent to urlencoded request.
README.md - added howto about afterFetch option, refresh event, parameter with function type.

v-zaburdaev commented:
> However, I would like you to make some changes before merging :
> define and document all the options you add (for instance disableGreenLines, showrawfcddata & afterFetch)

Removed disableGreenLines, showrawfcddata, and described afterFetch option. I need it to remove layers, it does not removed by this.clearLayers(); 

> use the same type of indentation as the master branch (2 spaces)
fixed.

> I would prefer that you convert the formdata to a urlencoded string or a json string and not the other way around as the FormData.append method can create arrays and direct assignation can't (around line 45)

At first my attemption was like you prefer, but it's dont work in safari brower, and some other browsers. https://developer.mozilla.org/en-US/docs/Web/API/FormData
Here is no suitable methods for all browsers, to read array from FormData.

> I would prefer not using a jQuery method for url encoding (as it adds a dependency)
removed jQuery dependency.

I add the "refresh" event, for refresh data, Maybe it's need to rename this event.

Thanks for you answer.
Vladimir.
